website/source/guides/security/vault-pki-integration.html.md

```diff
@@ -0,0 +1,550 @@
+
+Sign the intermediate CA CSR with the root certificate and save the generated
+certificate as `intermediate.cert.pem`:
+
+```shell
+$ vault write -format=json pki/root/sign-intermediate \
+    csr=@pki_intermediate.csr format=pem_bundle \
+    ttl="43800h" | jq -r '.data.certificate' > intermediate.cert.pem
+```
+
+Once the CSR is signed and the root CA returns a certificate, it can be imported
+back into Vault:
+
+```shell
+vault write pki_int/intermediate/set-signed certificate=@intermediate.cert.pem
+```
+
+### Step 6: Create a Role
+
+A role is a logical name that maps to a policy used to generate credentials. In
+our example, it will allow you to use [configuration
+parameters][config-parameters] that specify certificate common names, designate
+alternate names, and enable subdomains along with a few other key settings.
+
+Create a role named `nomad-cluster` that specifies the allowed domains, enables
+you to create certificates for subdomains, and generates certificates with a TTL
+of 86400 seconds (24 hours).
+
+```
+$ vault write pki_int/roles/nomad-cluster allowed_domains=global.nomad \
+    allow_subdomains=true max_ttl=86400s require_cn=false generate_lease=true
+```
+You should see the following output if the command you issues was successful:
+
+```
+Success! Data written to: pki_int/roles/nomad-cluster
+```
+
+### Step 7: Create a Policy to Access the Role Endpoint
+
+Recall from [Step 1](#step-1-initialize-vault-server) that we generated a root
+token that we used to log in to Vault. Although we could use that token in our
+next steps to generate our TLS certs, the recommended security approach is to
+create a new token based on a specific policy with limited privileges.
+
+Create a policy file named `tls-policy.hcl` and provide it the following
+contents:
+
+```
+path "pki_int/issue/nomad-cluster" {
+  capabilities = ["update"]
+}
+```
+Note that we have are specifying the `update` [capability][capability] on the
+path `pki_int/issue/nomad-cluster`. All other privileges will be denied. You can
+read more about Vault policies [here][policies].
+
+Write the policy we just created into Vault:
+
+```
+$ vault policy write tls-policy tls-policy.hcl
+Success! Uploaded policy: tls-policy
+```
+
+### Step 8: Generate a Token based on `tls-policy`
+
+Create a token based on `tls-policy` with the following command:
+
+```
+$ vault token create -policy="tls-policy" -ttl=24h
+```
+
+If the command is successful, you will see output similar to the following:
+
+```shell
+Key                  Value
+---                  -----
+token                s.xafiYzh7MCMotHLu2d35hepR
+token_accessor       9vj7q5nnF53JAcTyxvccpAZK
+token_duration       24h
+token_renewable      true
+token_policies       ["default" "tls-policy"]
+identity_policies    []
+policies             ["default" "tls-policy"]
+```
+
+Make a note of this token as you will need it in the upcoming steps.
+
+### Step 9: Configure Consul Template on All Nodes
```

Comment:
Should we flip these steps where it could be 1) create the /templates and /certs directories on each node first, then 2) explicitly create the templates and 3) then update the consul template 4) start the consul template service?